kraftwerk-core/src/main/java/fr/insee/kraftwerk/core/metadata/MetadataUtils.java

```diff
@@ -43,6 +43,12 @@ private static void putToMetadataVariable(String dataMode, ModeInputs modeInputs
 				addLunaticVariable(variables, filterResult, Constants.FILTER_RESULT_PREFIX, VariableType.BOOLEAN);
 			}
 		}
+		if (variables.getVariable("LIENS") != null) {
+			for (int k=1;k<21;k++) {
+				Variable varLien = new Variable("LIEN_"+k, variables.getGroup("BOUCLE_PRENOMS"), VariableType.INTEGER);
+				variables.putVariable(varLien);
+			}
+		}
```

Comment:
The hardcoded `"LIENS"`, `21`, `"LIEN_"`, `"BOUCLE_PRENOMS"` values should be refactor as static variable somewhere, this would also allow to put javadoc on these constants (from an outside perspective it's hard to understand this piece of code)

Reply:
I'm surprised to see "LIENS" and "BOUCLE_PRENOMS" hardcoded (these names comes from user-defined inputs in Pogues)